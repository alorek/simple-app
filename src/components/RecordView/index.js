import React, { Component } from 'react';
import { connect } from 'react-redux'

class RecordView extends Component {

    renderStandard(standard) {

        return (
            <div className="alert alert-info">
                <p>Standard: { standard.id }</p>
                <p>{ standard.description }</p>
            </div>
        );
    }

    render() {
    	const recordId = this.props.recordId;

    	const currentRecord = this.props.tableData.find(function(record) {
    		return record.id === recordId;
    	});

    	if (currentRecord) {
    		const standard = (currentRecord.standards || {}).standard;
            const mediaIcon = `../../assets/${currentRecord.media_type}.png`;
            const langIcon = `../../assets/${currentRecord.language}.png`;
            const langAlt = `Language: ${currentRecord.language}`;
            const mediaAlt = `Media Type: ${currentRecord.media_type}`;
            const cardStyle = {
                maxWidth: '500px'
            };
            const h4Style = {
                color: '#666666'
            };
            const topStyle = {
                 width: '100%'
            };
            const additionalTextStyle = {
                color: '#666666'
            };


	        return (

                <div className="card" style={ cardStyle }>
                    <div className="pull-left" style={ topStyle }>
                        <h4 className="pull-left" style={ h4Style }>
                            { currentRecord.categorization } | { currentRecord.resource_type }
                        </h4>
                        <div className="pull-right">
                            <img
                                height="24px"
                                width="24px"
                                src={ mediaIcon }
                                alt={ mediaAlt } />
                            <img
                                height="34px"
                                src={ langIcon }
                                alt={ langAlt }/>
                        </div>
                    </div>
                    <h3>{ currentRecord.display_title }</h3>
			        <p>{ currentRecord.meaningful_description }</p>
			        <p style={ additionalTextStyle }>{ currentRecord.additional_text }</p>
			        { standard && standard.description ? this.renderStandard(standard) : null }
                </div>
	        );
    	}

    	return (
	        <div><p>Record not found</p></div>
	    );
    }
}

function mapStateToProps(state) {
    return {
        tableData: state.table.tableData
    };
}

export default connect(mapStateToProps)(RecordView)
