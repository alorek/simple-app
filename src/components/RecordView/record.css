.record-wrapper {
    margin: 0 auto;
    max-width: 700px;
}

.record-wrapper .card {
    border: 1px solid #eee;
    box-shadow: 0 1px 3px rgba(0, 0, 0, .05);
    color: #444444;
    font-size: 16px;
    margin: 0 auto;
    max-width: 500px;
    padding: 15px;
}

.record-wrapper .card h4 {
    color: #666666;
}

.record-wrapper .card .top {
    width: 100%;
}
.record-wrapper .card .additional-text {
    color: #666666;
    font-size: 14px;
}
