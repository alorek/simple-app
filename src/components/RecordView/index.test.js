import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import RecordView from './index';
import reducer from '../../reducers';

const store = createStore(reducer);

test('RecordView renders correctly', () => {

    const tree = renderer.create(
        <Provider store={ store }>
            <RecordView  />
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
