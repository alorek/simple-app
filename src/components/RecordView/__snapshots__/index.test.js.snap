// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordView renders correctly 1`] = `
<div>
  <p>
    Not Found
  </p>
</div>
`;
